py/mkrules.mk

```diff
@@ -122,6 +122,14 @@ endif
 ifneq ($(PROG),)
 # Build a standalone executable (unix does this)
 
+COMPILER_TARGET := $(shell $(CC) -dumpmachine)
+ifneq (,$(findstring mingw,$(COMPILER_TARGET)))
```

Comment:
Would this be sufficient? I.e. suppose clang is used to target windows it will perhaps also add the extension? On the other hand since no problems have been reported this ight be ok for now. But from previous attempts it looks like this needs to be checked thoroughly. Do you think this covers at least all mingw cases including cross-compiling on unix for windows, and does it also work for mpy-cross?

Also see e.g. #3361 and #4143

Reply:
It does work for mpy-cross; this is the one that is currently broken under mingw. 

For some reason the mingw cross compiler does NOT add the .exe extension. I don't know what clang behavior is, but if I recall correctly in the past it used gcc's.

Reply:
In any case, adding the extension explicitly solves the issue, I think.

Reply:
@stinos this looks OK to me to merge, at least it seems to be the best fix so far (and there have been a lot of attempts).  There is precedence for this particular dumpmachine check, it's done already in mpy-cross/Makefile to test if fmode.c should be included or not.  What do you think?

Reply:
I assume this should be fine, and if it is then this part in appveyor.yml

```
    # Building of mpy-cross hasn't been fixed across all possible windows/WSL/...
    # variations and the STRIP step tries to strip mpy-cross whereas that should be
    # mpy-cross.exe. Workaround for now by skipping actual strip and size commands.
    C:\msys64\usr\bin\bash.exe -l -c "make -B -j4 V=1 STRIP=echo SIZE=echo"
```

can be adjusted as well